File: csharp/src/ev1/ut2_controlflujometodos/u04metodos/ejercicios/pendientes/E01_FuncionSaludo_Pendiente.cs
/*
OBJETIVO: Metodo void Saludar(string nombre); llamalo dos veces.
PATRON: Menu do-while con opcion 0 salir y uso de Console.ReadLine().
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/
using System;
class Program
{
    static void Main()
    {
        string? opcion;
        do
        {
            Console.WriteLine();
            Console.WriteLine("=== E01 Funcion Saludo Pendiente ===");
            Console.WriteLine("1. Preparar y revisar el ejercicio");
            Console.WriteLine("0. Salir");
            Console.Write("Elige una opcion: ");
            opcion = Console.ReadLine();
            switch (opcion)
            {
                case "1":
                    Console.WriteLine("Completa el metodo Saludar(string nombre) y usalo dos veces.");
                    Console.WriteLine("Prueba preparada: Saludar(\"Luis\") y Saludar(\"Sara\").");
                    Saludar("Luis");
                    Saludar("Sara");
                    break;
                case "0":
                    Console.WriteLine("Saliendo...");
                    break;
                default:
                    Console.WriteLine("Opcion no valida.");
                    break;
            }
        } while (opcion != "0");
    }
    static void Saludar(string nombre)
    {
        Console.WriteLine($"TODO: completar Saludar para mostrar un saludo a {nombre}.");
    }
}

File: cpp/src/ev1/ut2_controlflujometodos/u04metodos/ejercicios/pendientes/E01_FuncionSaludo_Pendiente.cpp
/*
OBJETIVO: Funcion void Saludar(string nombre); llamala dos veces.
PATRON: Menu do-while con opcion 0 salir y uso de cin.
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
#include <string>
using namespace std;

void ImprimirMenu() {
    cout << endl;
    cout << "=== E01 Funcion Saludo Pendiente ===" << endl;
    cout << "1. Preparar y revisar el ejercicio" << endl;
    cout << "0. Salir" << endl;
}

void Saludar(string nombre)
{
    cout << "TODO: completar Saludar para mostrar un saludo a " << nombre << "." << endl;
}

int main() {
    int opcion;
    do {
        ImprimirMenu();
        cout << "Introduce una opcion -> ";
        cin >> opcion;
        switch (opcion) {
            case 1:
                cout << "Completa la funcion Saludar(string nombre) y usala dos veces." << endl;
                cout << "Prueba preparada: Saludar(\"Luis\") y Saludar(\"Sara\")." << endl;
                Saludar("Luis");
                Saludar("Sara");
                break;
            case 0:
                cout << "Saliendo..." << endl;
                break;
            default:
                cout << "Opcion no valida." << endl;
                break;
        }
    } while (opcion != 0);
    return 0;
}
